### PR TITLE
if AppId already in URL, move faster (some fixes)

### PR DESCRIPTION
**Pull request by AngelRodriguez8008**
### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (updates to documentation, formatting, etc.)

### Description
Review PR #1085 to improve login. Check if AppId in the URL, don't try to open AppTiles directly.
Fix diferents URL missmatch. 
1. forceUCI=1 was not set on Switch App from Sales and Customer Service. (bug fix: Switch the App is not working)
2. forceUCI=1 was added between "&flags=easyreproautomation=true,testmode=true" and "&perf=true"

### Issues addressed
Improve the Login Performace, remove redudant Wait Call & Sleep time.
Add some tests for OpenApp.

### All submissions:
- [x] My code follows the code style of this project.
- [x] Do existing samples that are effected by this change still run?
- [x] I have added samples for new functionality. 
- [x] I raise detailed error messages when possible.
- [x] My code does not rely on labels that have the option to be hidden.

### Which browsers was this tested on?
- [x] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

**Comment by AngelRodriguez8008** [comment]
Working on: Switch the App is not working

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 1]
```diff
@@ -49,26 +51,50 @@ internal BrowserCommandResult<bool> InitializeModes()
...
-                if (Browser.Options.UCITestMode) queryParams += ",testmode=true";
-                if (Browser.Options.UCIPerformanceMode) queryParams += "&perf=true";
-
-                if (!uri.Contains(queryParams) && !uri.Contains(HttpUtility.UrlEncode(queryParams)))
+                string uri = driver.Url;
+                if (string.IsNullOrEmpty(uri))
+                    return false;
+                
+                var prevQuery = GetUrlQueryParams(uri);
+                bool requiereRedirect = false;
```
Comment (TYLEROL) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: nit: requireRedirect

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 2]
```diff
@@ -49,26 +51,50 @@ internal BrowserCommandResult<bool> InitializeModes()
...
+                string queryParams = "";
+                if (prevQuery.Get("flags") == null)
                 {
-                    var testModeUri = uri + queryParams;
+                    queryParams+= "&flags=easyreproautomation=true";
+                    if (Browser.Options.UCITestMode)
+                        queryParams += ",testmode=true";
+                    requiereRedirect = true;
```
Comment (TYLEROL) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: nit: requireRedirect

**Review by TYLEROL — CHANGES_REQUESTED** [review]

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs (hunk 2 above)
Comment (AngelRodriguez8008) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: done

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs (hunk 1 above)
Comment (AngelRodriguez8008) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: done

**Review by AngelRodriguez8008 — COMMENTED** [review]
Typo fixed

**Review by TYLEROL — APPROVED** [review]

**Pull request merged**